[WIP] Release PR for v3.5.0 (#6454)
redhat-developer/odo: docs/website/blog/2023-01-04-odo-v3.5.0.md

---
title: odo v3.5.0 Released
author: Philippe Martin
author_url: https://github.com/feloy
author_image_url: https://github.com/feloy.png
tags: ["release"]
slug: odo-v3.5.0
---

odo `v3.5.0` is now out!

<!--truncate-->

To install `odo`, follow [the installation guide](../docs/overview/installation).

## Notable Changes
Check this Playlist for an overview of the most notable changes in this release: [odo v3.5.0 (Sprint 229)](https://www.youtube.com/playlist?list=PLGMB2PY4SNOoGCUFCbqxnQclhnHA0L73D)

### Features

#### `odo delete component` deletes component running on podman

> When you run `odo dev` on podman platform, pressing ctrl-c will delete the resources from podman. 
> 
> If `odo dev` does not delete the reosurces as expected (because it failed unexpectedly, it has been killed with the KILL(9) signal, etc), you can use `odo delete component` to delete the resources. 

Reviewer: ```suggestion
When you run `odo dev` on podman platform, pressing Ctrl+c will delete the resources from podman. If `odo dev` does not delete the resources as expected (because it failed unexpectedly, it has been killed with the KILL(9) signal, etc), you can use `odo delete component` to delete the resources. 
```

I find the extra line less readable. I would prefer we remove it.

> Note that you need to run odo in the experimental mode for this feature. 

Reviewer: ```suggestion
Note that you need to run delete command in the experimental mode for this feature.
```

<iframe width="560" height="315" src="https://www.youtube.com/embed/ezLUArXMaGs" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>

#### `odo describe component` describes component running on podman

You can now get the description of a component running on podman with the command `odo describe component`.

> Note that you need to run odo in the experimental mode for this feature. 

Reviewer: ```suggestion
Note that you need to run the describe command in the experimental mode for this feature.
```

If `odo dev` does not delete the reosurces as expected (because it failed unexpectedly, it has been killed with the KILL(9) signal, etc), you can use `odo delete component` to delete the resources. 

Note that you need to run odo in the experimental mode for this feature. 

<iframe width="560" height="315" src="https://www.youtube.com/embed/ezLUArXMaGs" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>

#### `odo describe component` describes component running on podman

You can now get the description of a component running on podman with the command `odo describe component`.

Note that you need to run odo in the experimental mode for this feature. 

<iframe width="560" height="315" src="https://www.youtube.com/embed/hhEqPfLurzI" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>

#### Allow fetching a specific Devfile Stack version

> Version 3.4.0 of odo introduced the support of versioned Devfile stacks, with the ability to list available versions od Devfile stacks. Starting with version 3.5.0, you can initialize a new project with a specific version of a Devfile stack. Previously, the default version were used.

Reviewer 2: ```suggestion
Version 3.4.0 of odo introduced the support of versioned Devfile stacks, with the ability to list available versions of Devfile stacks. Starting with version 3.5.0, you can initialize a new project with a specific version of a Devfile stack. Previously, the default version was used.
```
